core/templates/dev/head/services/CsrfTokenService.ts

```diff
@@ -15,6 +15,9 @@
 /**
  * @fileoverview Service for managing CSRF tokens.
  */
+
+import * as $ from 'jquery';
```

Comment:
This is a bit concerning. What's going on here?

Reply:
This is actually a fix to something not going well with Karma. After the introduction of Angular and upgrading some services, this service started to come under Angular zone due to it being dependency, now the zone was not able to find AJAX directly as a global like we do in AngularJS (the same reason we cannot upgrade the constants like `INTERACTION_SPECS` which are defined in a global variable).

Reply:
And hence we need to import it specifically, like we do for all the Angular things (like the imports in AppInit.ts are explicit and do not depend on global definitions)

Reply:
There needs to be a comment for things like this, which explains what is going on and that points to an authoritative link explaining the context behind such a solution.

Reply:
Okay I will add a comment to it.
But about a link, I do not have one because I discovered this fix myself. There is [one open question](https://stackoverflow.com/questions/49252655/injecting-lodash-in-karma/57074641#57074641) in which the problem of OP is the same as mine but he also did not get a correct answer, I myself have posted an answer on it which was this very fix. I don't know which link to provide.

Could you please give some suggestions?

Reply:
You can link to that thread. Also, explain in the comment what this syntax is actually doing.

(Thanks for helping other folks on StackOverflow, btw!)

Reply:
Done!